jerry-ext/arg/arg.c

```diff
@@ -81,8 +81,51 @@ jerryx_arg_transform_this_and_args (const jerry_value_t this_val, /**< the this_
   {
     jerry_release_value (ret);
 
-    return jerry_create_error (JERRY_ERROR_TYPE, (jerry_char_t *) "'this' validation failed");
+    return jerry_create_error (JERRY_ERROR_TYPE, (jerry_char_t *) "'this' validation failed.");
   }
 
   return jerryx_arg_transform_args (js_arg_p, js_arg_cnt, c_arg_p + 1, c_arg_cnt - 1);
 } /* jerryx_arg_transform_this_and_args */
+
+/**
+ * Validate the `obj_val`'s properties,
+ * and assign them to the native arguments.
+ *
+ * @return jerry undefined: all validators passed,
+ *         jerry error: a validator failed.
+ */
+jerry_value_t
+jerryx_arg_transform_object_properties (const jerry_value_t obj_val,/**< the JS object */
+                                        const jerry_char_t **name_p, /**< property name list of the JS object */
+                                        const jerry_length_t name_cnt, /**< count of the name list */
+                                        const jerryx_arg_t *c_arg_p, /**< points to the array of transformation steps */
+                                        jerry_length_t c_arg_cnt) /**< the count of the `c_arg_p` array */
+{
+  if (!jerry_value_is_object (obj_val))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE, (jerry_char_t *) "Not an object.");
+  }
+
+  jerry_value_t prop[name_cnt];
+
+  for (jerry_length_t i = 0; i < name_cnt; i++, name_p++)
+  {
+    const jerry_value_t name_str = jerry_create_string (*name_p);
+    prop[i] = jerry_get_property (obj_val, name_str);
+    jerry_release_value (name_str);
+
+    if (jerry_value_has_error_flag (prop[i]))
+    {
```

Comment:
If an error happened at `i == N`, you need to release `prop[0..N-1]` before returning.